scimath_f/lawson.f

```diff
@@ -290,7 +310,7 @@ SUBROUTINE NNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
       DIMENSION A(MDA,N), B(1), X(1), W(1), ZZ(1)
       INTEGER INDEX(1)
       REAL      A, B, X, RNORM, W, ZZ, ZERO, ONE, TWO, FACTOR, SM,
-     $   WMAX, ASAVE, UP, DUMMY, UNORM, ZTEST, ALPHA, T, CC, SS
+     $   WMAX, ASAVE, UP, DUMMY(1), UNORM, ZTEST, ALPHA, T, CC, SS
```

Comment:
If this is the same dummy that is declared as double precision below, it is contradictory.

Reply:
Aren't they in different subroutines?

Reply:
Ah it's in different functions, so is fine.